lesson_37/first-react-app/src/components/main.jsx

```diff
@@ -1,44 +1,37 @@
 import React from "react"
 import ReactDOM from "react-dom/client"
-import {useState} from "react"
+import {useState, useEffect} from "react"
 import "../../node_modules/bootstrap/dist/css/bootstrap.css"
 import "../../node_modules/bootstrap/dist/js/bootstrap.js"
 import Header from "./Header.jsx"
 import Products from "./Products.jsx"
+import CreateProduct from "./CreateProduct.jsx"
 import "./index.css"
 
+export const NotificationContext = React.createContext()
+
 export const App = () => {
-  let tv = {
-    display: 65,
-    model: "SAMSUNG",
-    color: "white",
-    price: 250,
-    img: "src/assets/TV.jpeg"
-  }
-  let laptop = {
-    display: 15.6,
-    model: "ACER",
-    color: "black",
-    price: 500,
-    img: "src/assets/laptop.jpeg"
-  }
-  let xBox = {
-    display: "TV",
-    model: "XBOX",
-    color: "black",
-    price: 350,
-    img: "src/assets/X-Kom.jpeg"
-  }
-  let products = [tv, laptop, xBox]
-  const [productsIn, setProducts] = useState(products)
+  const [productsIn, setProducts] = useState([])
   const [productsInCart, setProductsInCart] = useState([])
 
+  useEffect(() => {
+    fetch("https://dummyjson.com/products")
+      .then((response) => response.json())
+      .then((data) => {
+        let fetchProducts = data.products
+        setProducts(fetchProducts)
+      })
+  }, [])
+
   const addToCart = (prod) => {
     setProductsInCart([...productsInCart, prod])
   }
   return (
     <React.StrictMode>
       <Header count={productsInCart.length}></Header>
+      <NotificationContext.Provider value={[productsIn, setProducts]}>
+        <CreateProduct></CreateProduct>
+      </NotificationContext.Provider>
       <Products productsIn={productsIn} addToCart={addToCart}></Products>
```

Comment:
why this component is outside of Context?
if you pass `productsIn` by param there is no benefits of context then